migrations/src/main/scala/za/co/absa/enceladus/migrations/continuous/EntityVersionMapMongo.scala

```diff
@@ -0,0 +1,121 @@
+
+  /**
+    * An implementation if this abstract class should redefine this method. Clients of this class should use add()
+    * as it does additional checks.
+    *
+    * @param collectionName The name of the collection that contains the entity
+    * @param entityName     An entity name
+    * @param oldVersion     An version of the entity in the old version of the database
+    * @param newVersion     An version of the entity in the new version of the database
+    */
+  override protected def addEntry(collectionName: String,
+                                  entityName: String,
+                                  oldVersion: Int,
+                                  newVersion: Int): Unit = {
+    getTypedCollection[EntityVersionMapping]
+      .insertOne(EntityVersionMapping(collectionName, entityName, oldVersion, newVersion))
+      .execute()
+  }
+
+  /**
+    * Gets a 'name - version' mapping.
+    *
+    * @param collectionName The name of the collection that contains the entity
+    * @param entityName     An entity name
+    * @param oldVersion     An version of the entity in the old version of the database
+    * @return An version of the entity in the new version of the database, None if the entity is not found
+    *         in the mapping
+    */
+  @throws[IllegalStateException]
+  override def get(collectionName: String,
+                   entityName: String,
+                   oldVersion: Int): Option[Int] = {
+    val mappings = getTypedCollection[EntityVersionMapping]
+      .find(getFilter(collectionName, entityName, oldVersion))
+      .execute()
+    if (mappings.lengthCompare(1) > 0) {
+      throw new IllegalStateException("Duplicate entries in entity version mappings found. " +
+        s"Values: $mappings.")
+    } else {
+      mappings.headOption.map(record => record.newVersion)
+    }
+  }
+
+  /**
+    * Returns a collection that is serialized as a case class.
+    */
+  private def getTypedCollection[T](implicit ct: ClassTag[T]): MongoCollection[T] = {
+    db.getCollection[T](entityMapCollectionName)
+      .withCodecRegistry(codecRegistry)
+  }
+
+  /**
+    * Returns a filter for the MongoDB query for finding entity version mapping.
+    */
+  private def getFilter(collectionName: String,
+                        entityName: String,
+                        oldVersion: Int): Bson = {
+    and(
+      equal("collection", collectionName),
+      equal("entityName", entityName),
+      equal("oldVersion", oldVersion))
+  }
+
+  /**
+    * Ensures that the entity version mapping collections exists
+    */
+  private def ensureCollectionExists(): Unit = {
+    val collections = db.listCollectionNames().execute()
+    if (!collections.contains(entityMapCollectionName)) {
+      db.createCollection(entityMapCollectionName).execute()
+      db.getCollection(entityMapCollectionName)
+        .createIndex(
+          Document(
+            Seq("collection" -> 1, "entityName" -> 2, "oldVersion" -> 3)
```

Comment:
```suggestion
            Seq("collection" -> 1, "entityName" -> 1, "oldVersion" -> 1)
```
It's standard practice to specify `1 (ASC)` and `-1 (DESC)`, let's not use other values.